src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,73 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert shoppingCart" + shoppingCart, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> getShoppingCartByUser =
+                    session.createQuery("select distinct sc "
+                            + "FROM  ShoppingCart as sc"
+                            + " left join  fetch sc.tickets "
+                            + "WHERE sc.id = :id ", ShoppingCart.class);
```

Comment:
```suggestion
            Query<ShoppingCart> getShoppingCartByUser =
                    session.createQuery("SELECT DISTINCT sc "
                            + "FROM  ShoppingCart AS sc"
                            + " LEFT JOIN  FETCH sc.tickets "
                            + "WHERE sc.id = :id ", ShoppingCart.class);
```
використовуй або uppercase або lowercase, не змішуй